### PR TITLE
Revert "Revert "Symintifying slice ops (#85196)""

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #85746

This reverts commit 3a171dfb0c08956d55f341039cf35e3a18269c34.

### CODE REVIEW

**Review by albanD — APPROVED** [review]
:crossed_fingers:

**Comment by ezyang** [comment]
There are some new torchgen changes here, mostly just making it possible to turn off symint codegen for executorch.

File: aten/src/ATen/test/vulkan_api_test.cpp [hunk 1]
```diff
@@ -173,7 +173,8 @@ static void slice_tests(const std::unordered_map<int64_t, std::vector<int64_t>>&
     slice_test(dim2size.second, dim2size.first, -30, -10, 1);       // i.e., 4D tensor's equivalent indexing = [:,:,:,-30:-10:1] with negative start/end
     slice_test(dim2size.second, dim2size.first, 0, INT64_MAX, 1);   // i.e., 4D 's equivalent indexing = [:,:,:,0:9223372036854775807:1] with end=INT64_MAX
     slice_test(dim2size.second, dim2size.first, -10, INT64_MAX, 1); // i.e., 4D 's equivalent indexing = [:,:,:,-10:9223372036854775807:1] with negative start and end=INT64_MAX
-    slice_test(dim2size.second, dim2size.first, INT64_MIN, INT64_MAX, 1); // i.e., 4D 's equivalent indexing = [:,:,:,-9223372036854775808:9223372036854775807:1] with start=INT64_MIN and end=INT64_MAX
+    // This triggers a SymInt assert since [-2^63, -2^62-1] range is reserved for packed symints
```
Comment (albanD) [inline aten/src/ATen/test/vulkan_api_test.cpp:176]: lol

File: test/lazy/test_ts_opinfo.py [hunk 2]
```diff
@@ -162,7 +173,7 @@ def get_name(op):
                 l.append(op.variant_test_name)
             return '.'.join(l)
 
-        global FALLBACK_LIST
+        global HAS_SYMINT_SUFFIX, FALLBACK_LIST
```
Comment (albanD) [inline test/lazy/test_ts_opinfo.py:176]: This is not needed as these variables are not modified.

**Review by albanD — APPROVED** [review]
The disable sounds good.

**Pull request closed without merging**